### PR TITLE
lua: Add Windows support

### PR DESCRIPTION
**Pull request by sakloui**
The current version of the extension tries to download the Windows binary files for lua-language-server like this:
`lua-language-server-3.9.3-win32-x64.tar.gz`

The [Windows binary files](https://github.com/LuaLS/lua-language-server/releases) are only released as zip archives, so it will fail to get the required files. 

This pr changes the following:
- Add check for Windows specific zip archive
- Add check for Windows specific .exe executable

Release Notes:

- N/A

### CODE REVIEW

File: extensions/lua/src/lua.rs [hunk 1]
```diff
@@ -58,7 +63,14 @@ impl LuaExtension {
...
         let version_dir = format!("lua-language-server-{}", release.version);
-        let binary_path = format!("{version_dir}/bin/lua-language-server");
+        let binary_path = format!(
+            "{version_dir}/bin/lua-language-server{}",
+            match platform {
+                zed::Os::Mac => "",
+                zed::Os::Linux => "",
+                zed::Os::Windows => ".exe",
+            },
+        );
```
Comment (JunkuiZhang) [inline extensions/lua/src/lua.rs:73]:
```suggestion
        let binary_path = format!("{version_dir}/bin/lua-language-server{}", std::env::consts::EXE_SUFFIX);
```

Comment (sakloui) [reply extensions/lua/src/lua.rs:73]: `std::env::consts::EXE_SUFFIX` returns `.wasm` and not `.exe` unfortunately

Comment (JunkuiZhang) [reply extensions/lua/src/lua.rs:73]: Oh right, we are under wasm env. Sorry.

Comment (sakloui) [reply extensions/lua/src/lua.rs:73]: No problem. My knowledge of Rust is quite minimal and your change would be a lot nicer!

**Review by maxdeviant — APPROVED** [review]
Thank you!

**Pull request merged**